src/wp-includes/blocks.php

```diff
@@ -1880,7 +1881,41 @@ function resolve_pattern_blocks( $blocks ) {
 				continue;
 			}
 
-			$blocks_to_insert   = parse_blocks( $pattern['content'] );
+			$blocks_to_insert = parse_blocks( trim( $pattern['content'] ) );
+
+			$blocks_to_insert = parse_blocks( trim( $pattern['content'] ) );
+
+			/*
+			* For single-root patterns, add the pattern name to make this a pattern instance in the editor.
+			* If the pattern has metadata, merge it with the existing metadata.
+			*/
+			if ( count( $blocks_to_insert ) === 1 ) {
+				$block_metadata                = $blocks_to_insert[0]['attrs']['metadata'] ?? array();
+				$block_metadata['patternName'] = $slug;
+
+				/*
+				* Merge pattern metadata with existing block metadata.
+				* Pattern metadata takes precedence, but existing block metadata
+				* is preserved as a fallback when the pattern doesn't define that field.
+				* Only the defined fields (name, description, categories) are updated;
+				* other metadata keys are preserved.
+				*/
+				foreach ( array(
+					'name'        => 'title', // 'title' is the field in the pattern object 'name' is the field in the block metadata.
+					'description' => 'description',
+					'categories'  => 'categories',
+				) as $key => $pattern_key ) {
+					$value = $pattern[ $pattern_key ] ?? $block_metadata[ $key ] ?? null;
+					if ( $value ) {
+						$block_metadata[ $key ] = 'categories' === $key && is_array( $value )
```

Comment:
Is the `$key` check needed here? Shouldn't `array_map()` always be used when the value is an array, regardless of the key?

```suggestion
						$block_metadata[ $key ] = is_array( $value )
```

Reply:
By jove, you're right! I just commit stuff when it works